animations/lib/src/misc/animated_list.dart

```diff
@@ -93,31 +94,23 @@ class _AnimatedListDemoState extends State<AnimatedListDemo> {
 }
 
 class UserModel {
-  const UserModel({this.firstName, this.lastName});
+  UserModel(
+    this.id,
+    this.firstName,
+    this.lastName,
+  );
 
+  final int id;
   final String firstName;
   final String lastName;
 }
 
 List<UserModel> initialListData = [
-  UserModel(
-    firstName: 'Govind',
-    lastName: 'Dixit',
-  ),
-  UserModel(
-    firstName: 'Greta',
-    lastName: 'Stoll',
-  ),
-  UserModel(
-    firstName: 'Monty',
-    lastName: 'Carlo',
-  ),
-  UserModel(
-    firstName: 'Petey',
-    lastName: 'Cruiser',
-  ),
-  UserModel(
-    firstName: 'Barry',
-    lastName: 'Cade',
-  ),
+  UserModel(0, 'Govind', 'Dixit'),
+  UserModel(1, 'Greta', 'Stoll'),
+  UserModel(2, 'Monty', 'Carlo'),
+  UserModel(3, 'Petey', 'Cruiser'),
+  UserModel(4, 'Barry', 'Cade'),
 ];
+
+int _maxIdValue = 4;
```

Comment:
Could this be based off the length of the `listData` list instead?

Reply:
I believe that would cause problems if items were repeatedly added and removed. For example:

1) Add a new UserModel, which would get an ID of 5 if the code's using the length of the list
2) Remove the first UserModel in the list.
3) Add another UserModel, which would also get an ID of 5.

Reply:
ah, right.